### DISCUSSION TITLE
show empty sections in prompt

### DISCUSSION THREAD

**Discussion by VPraharsha03** [JanDeDobbeleer/oh-my-posh#4779] [General]
I'm trying to port a starship theme to OMP, I'm trying to figure out if there's a way to keep empty sections when there are no related files of that particular module as shown below

![20240315_231304.jpg](https://github.com/JanDeDobbeleer/oh-my-posh/assets/56556436/bd47aab5-a5f2-43c0-8c77-2db6a22b4479)

**Comment by JanDeDobbeleer** [comment]
@VPraharsha03 if you change them from `powerline` to `accordion`, this will do exactly that out of the box. See [here](https://ohmyposh.dev/docs/configuration/segment#accordion).

**Reply by VPraharsha03** [reply]
@JanDeDobbeleer but here's a small catch, as you can observe the time segment has an end symbol, accordion and powerline seem to only allow specifying the start symbol... is there a way to do this

**Reply by JanDeDobbeleer** [reply]
@VPraharsha03 time needs to be a diamond with only a trailing diamond.

**Reply by VPraharsha03** [reply]
> @VPraharsha03 time needs to be a diamond with only a trailing diamond.

should've mentioned my bad, i'm making a small change - the last segment except time should've a rounded end and should be shown even when empty 

time segment would move to the right end

**Reply by VPraharsha03** [reply]
is there a way to make a diamond behave like an accordion or style an accordion ? @JanDeDobbeleer

**Reply by JanDeDobbeleer** [reply]
Not yet. I assume that last one to always be enabled here, no?

**Reply by JanDeDobbeleer** [reply]
@VPraharsha03 can you show me a few screenshots with use-cases? That way I can help you either define how to, or create a new feature that supports this.

**Reply by VPraharsha03** [reply]
@JanDeDobbeleer 
I'm using a slightly modified version of the starship prompt with the time segment shifted to the right corner and the last segment on left behaving like an accordion with rounded end

![omp.png](https://github.com/JanDeDobbeleer/oh-my-posh/assets/56556436/d13bb373-ba71-4188-8d61-65e3a7f32a1d)

**Reply by JanDeDobbeleer** [reply]
Aha.

**Reply by VPraharsha03** [reply]
one more thing was that I'm unable to always show node version despite changing the "display_node" to `always` in "properties" section

**Reply by JanDeDobbeleer** [reply]
@VPraharsha03 you need to also enable `fetch_version`.

**Reply by VPraharsha03** [reply]
@JanDeDobbeleer `fetch_version` always defaults to `true` it seems, for me I'd to enable `home_enabled` to true in order to see it when the current directory is ~/

**Reply by JanDeDobbeleer** [reply]
Ah yes, forgot about that one indeed.

**Reply by VPraharsha03** [reply]
> 
> ![omp.png](https://github.com/JanDeDobbeleer/oh-my-posh/assets/56556436/d13bb373-ba71-4188-8d61-65e3a7f32a1d)
> 
> 

@JanDeDobbeleer how can I make the time segment self adjust it's position on window resizing ?

**Reply by JanDeDobbeleer** [reply]
@VPraharsha03 you can't. Shells don't offer that functionality.

**Reply by VPraharsha03** [reply]
@JanDeDobbeleer oh, one thing that I tried was to change it to rprompt but then the time segment would appear in the right side of input prompt (not in the same line as other segments but in the next line below)

**Reply by JanDeDobbeleer** [reply]
@VPraharsha03 resizing will always ruin the prompt rendering.

**Reply by VPraharsha03** [reply]
@JanDeDobbeleer yeah, i've seen other weird rendering issues (the time segment getting elongated all the way into next line on restoring), thanks for the info.

**Comment by VPraharsha03** [comment]
Is there a generic `packages` module/segment to display package versions of npm, cargo, dart etc.. ? @JanDeDobbeleer

**Reply by JanDeDobbeleer** [reply]
@VPraharsha03 you're looking for `project`
